### PR TITLE
implement DraftConverter.convertToApiData

### PR DESCRIPTION
**Pull request by nickhsine**
### REFERENCE
spec: https://github.com/twreporter/keystone/issues/16

### code change
1. add ApiDataInstance.js to define the api data structure
2. add AtomicBlockProcessor.js for 'media' and 'atomic' block type
3. store apiData into mongoDB by fields/types/html/HtmlType.js

### TODO

- only original image has ```width``` and ```height``` property.  ```desktop```, ```mobile``` and ```tablet``` only has ```id```, ```description``` and ```url```. We should store all all of them(```desktop```, ```mobile```, ```tablet``` and ```original```) while uploading the image at first.

@hcchien @garfieldduck

### CODE REVIEW

File: fields/types/html/AtomicBlockProcessor.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        });
+    });
+    }
+    */
+
+function getResizedImageUrls(imageUrl) {
+    let desktop = '';
+    let mobile = '';
+    let tablet = '';
+    if (typeof imageUrl === 'string' && imageUrl) {
```
Comment (hcchien) [inline fields/types/html/AtomicBlockProcessor.js]: do you mean
(imageUrl && typeof imageUrl === 'string')
?

Comment (nickhsine) [inline fields/types/html/AtomicBlockProcessor.js]: it is the same thing.

File: fields/types/html/AtomicBlockProcessor.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        const matched = imageUrl.match(/(\.[^.]+)?$/);
```
Comment (hcchien) [inline fields/types/html/AtomicBlockProcessor.js]: TODO:
add the url pattern check

File: fields/types/html/DraftConverter.js [hunk 3]
```diff
@@ -77,8 +75,52 @@ function _convertBlocksToHtml(blocks, entityMap) {
...
+                content.push(converted);
+            } else if (nestLevel[0] === block.type) {
+                // previous block is a item-list and current block is the same item-list
+                content.push(converted);
+            } else if (nestLevel[0] !== block.type) {
+                // previous block is a different item-list.
+                apiDataArr = apiDataArr.push(new ApiDataInstance({type: nestLevel[0], content: [content]}));
+                content = [converted];
+                nestLevel.shift();
+                nestLevel.unshift(block.type);
```
Comment (nickhsine) [inline fields/types/html/DraftConverter.js]: nestLevel[0] = block.type;

**Pull request merged**